lib/magma/server/update.rb

```diff
@@ -22,6 +22,8 @@ def action
 
     load_revisions if success?
 
+    update_any_file_links if success?
+
```

Comment:
I think the metis file stuff must happen before the load request to the database, because if metis replies with a 403 or something, Magma should not perform the db update. On the other hand, the load itself sometimes fails because of poor upstream validation, resulting in inserting the wrong type, duplicate keys, etc. Hard to know how to resolve this conundrum.

Reply:
Hm....interesting. Will noodle on it.